File: Group-Anagrams.java
1class Solution {
2    public List<List<String>> groupAnagrams(String[] strs) {
3        HashMap <String, List<String>> map = new HashMap <> ();
4        for (String s:strs)
5        {
6            char[] chars = s.toCharArray();
7            Arrays.sort(chars);
8            String key = new String(chars);
9            if(!map.containsKey(key))
10            {
11                map.put(key,new ArrayList<>());
12            }
13            map.get(key).add(s);
14        }
15        return new ArrayList<>(map.values());
16    }
17}

File: Group-Anagrams.py
1class Solution:
2    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
3        ans = defaultdict(list)
4        for word in strs:
5            key = "".join(sorted(word))
6            ans[key].append(word)
7        return list(ans.values())
8